asyncscope.md

```diff
@@ -2306,6 +2301,401 @@ template <async_scope_token Token, sender Sender>
+    };
+
+template <class Token>
+concept async_scope_token =
+    copyable<Token> &&
+    requires(Token token) {
+        { token.try_associate() } -> async_scope_association;
+        { token.wrap(declval<@_test-sender_@>()) } -> sender_in<@_test-env_@>;
+    };
+
+struct simple_counting_scope::assoc {
+    explicit operator bool() const noexcept;
+
+private:
+    simple_counting_scope* @_scope_@; // exposition only
+};
+
+struct simple_counting_scope::token {
+    template <sender Sender>
+    Sender&& wrap(Sender&& snd) const noexcept;
+
+    assoc try_associate() const noexcept;
+
+private:
+    simple_counting_scope* @_scope_@; // exposition only
+};
+
+
+struct counting_scope::assoc {
+    explicit operator bool() const noexcept;
+
+private:
+    counting_scope* @_scope_@; // @_exposition only_@
+};
+
+struct counting_scope::token {
+    template <sender Sender>
+    sender auto wrap(Sender&& snd) const noexcept;
+
+    assoc try_associate() const noexcept;
+
+private:
+    counting_scope* @_scope_@; // @_exposition only_@
+};
+
+}
+```
+::::
+
+:::cmptable
+### Before
+```cpp
+namespace std::execution {
+
+template <class Alloc, async_scope_token Token, sender Sender>
+struct @_spawn-state_@ : @_spawn-state-base_@ {
+    using @_op-t_@ = connect_result_t<Sender, @_spawn-receiver_@>; // @_exposition only_@
+
+    @_spawn-state_@(Alloc a, Sender&& sndr, Token t)
+        : @_alloc_@(std::move(a)),
+          @_op_@(connect(std::move(sndr), @_spawn-receiver_@{this})),
+          @_token_@(std::move(t)) {}
+
+    void @_run_@() {
+        if (@_token_@.try_associate())
+            @_op_@.start();
+        else
+            @_destroy_@();
+    }
+    void @_complete_@() noexcept(is_nothrow_move_constructible_v<Token>) override {
+        auto t = std::move(@_token_@);
+
+        @_destroy_@();
+
+        t.disassociate();
+    }
+
+private:
+    using @_alloc-t_@ = // @_exposition only_@
+        typename allocator_traits<Alloc>::template rebind_alloc<@_spawn-state_@>;
+
+    @_alloc-t_@ @_alloc_@; // @_exposition only_@
+    @_op-t_@ @_op_@; // @_exposition only_@
+    Token @_token_@; // @_exposition only_@
+
+    void @_destroy_@() noexcept { // @_exposition only_@
+        auto a = std::move(@_alloc_@);
+
+        allocator_traits<@_alloc-t_@>::destroy(a, this);
+        allocator_traits<@_alloc-t_@>::deallocate(a, this, 1);
+    }
+
+};
+
+}
+```
+
+### After
+```cpp
+namespace std::execution {
+
+template <class Alloc, async_scope_token Token, sender Sender>
+struct @_spawn-state_@ : @_spawn-state-base_@ {
+    using @_op-t_@ = connect_result_t<Sender, @_spawn-receiver_@>; // @_exposition only_@
+    using @_assoc-t_@ = remove_cvref_t<decltype(declval<Token&>().try_associate())>; // @_exposition only_@
+
+    @_spawn-state_@(Alloc a, Sender&& sndr, Token t)
+        : @_alloc_@(std::move(a)),
+          @_op_@(connect(std::move(sndr), @_spawn-receiver_@{this})) {
+        @_assoc_@ = t.try_associate();
+    }
+
+    void @_run_@() {
```

Comment:
This can be `noexcept` because all the constituent ops are. Also, it's expo-only.
```suggestion
    void @_run_@() noexcept { // exposition only
```